### ISSUE TITLE
App crashes on startup after not rebooting for some time

### ISSUE THREAD

**Issue by ActualDisaster2447** [hezral/clips#43]
I've noticed that Clips crashes whenever I launch it after a day or two of not rebooting. The crash happens if I launch it via the keyboard shortcut or through the application menu - it doesn't matter. Unfortunately, I don't know what logs I could provide to accelerate solving the issue, but I'm more than happy to provide any on request.

Thanks in advance!

**Comment by hezral** [comment]
Thanks for reporting. 

Can you share how many clips you have now. Also if you can reproduce this by command line ie running same command you put for the keyboard shortcut.

**Comment by ActualDisaster2447** [comment]
For some reason, Clips has been working fine for the most part, although, sometimes it just decides to crash. I have 49 clips right now and when launching to check it crashed. Then when launching it through terminal, it didn't crash. Is there a directory where logs are saved so that I can upload them here? Because I can't launch Clips from terminal each time to check for logs in case it crashes again.

Thank you for responding by the way!

**Comment by ActualDisaster2447** [comment]
Somehow I kept trying to launch it from terminal until it crashed (tried two times until it crashed on the third time):

```
$ flatpak run com.github.hezral.clips 
Clips 1.0.9
2022-02-01 14:33:41.048740 startup
Gtk-Message: 14:33:41.070: Failed to load module "canberra-gtk-module"
2022-02-01 14:33:41.073872 open_db
Found ClipsDB [(0, 'main', '/home/muffin/.var/app/com.github.hezral.clips/cache/com.github.hezral.clips/com.github.hezral.clips.db')]
2022-02-01 14:33:41.084529 file manager backend started
2022-02-01 14:33:41.336579 start auto-housekeeping
2022-02-01 14:33:41.336644 auto-retention-period 365
0 found for auto housekeeping
No records found for auto housekeeping
2022-02-01 14:33:41.336794 finish auto-housekeeping
2022-02-01 14:33:41.339530 start load_clips
2022-02-01 14:33:42.809077 finish load_clips
2022-02-01 14:33:41.340137 execution time of load_clips_fromdb: 1.468972 sec

(com.github.hezral.clips:2): GdkPixbuf-CRITICAL **: 14:33:45.209: gdk_pixbuf_new_subpixbuf: assertion 'src_y >= 0 && src_y + height <= src_pixbuf->height' failed
Traceback (most recent call last):
  File "/app/share/com.github.hezral.clips/clips/clips_view.py", line 1088, in draw
    Gdk.cairo_set_source_pixbuf(cairo_context, final_pixbuf, x, y)
TypeError: Argument 1 does not allow None as a value

(com.github.hezral.clips:2): GdkPixbuf-CRITICAL **: 14:33:45.227: gdk_pixbuf_new_subpixbuf: assertion 'src_y >= 0 && src_y + height <= src_pixbuf->height' failed
Traceback (most recent call last):
  File "/app/share/com.github.hezral.clips/clips/clips_view.py", line 1088, in draw
    Gdk.cairo_set_source_pixbuf(cairo_context, final_pixbuf, x, y)
TypeError: Argument 1 does not allow None as a value
python3: ../../src/cairo.c:524: cairo_destroy: Assertion `CAIRO_REFERENCE_COUNT_HAS_REFERENCE (&cr->ref_count)' failed.
```

**Comment by hezral** [comment]
Thanks, that narrows it a bit. I haven't tried stress testing it that way 😅 let me try to reproduce. 

Currently it doesn't output to a log file as but i am planning to include a debug mode in next version

**Comment by ActualDisaster2447** [comment]
Thank you for your effort solving this. Once the update with the debug mode is available, I will provide more logs for sure (if necessary).

**Comment by BepBop** [comment]
`➜  ~ flatpak run com.github.hezral.clips 
Clips 1.0.9
2022-02-28 22:16:23.288133 startup
Gtk-Message: 22:16:23.324: Failed to load module "canberra-gtk-module"
2022-02-28 22:16:23.330163 open_db
Found ClipsDB [(0, 'main', '/home/acer/.var/app/com.github.hezral.clips/cache/com.github.hezral.clips/com.github.hezral.clips.db')]
2022-02-28 22:16:23.351048 file manager backend started
2022-02-28 22:16:23.649439 start load_clips

(com.github.hezral.clips:2): GdkPixbuf-CRITICAL **: 22:16:23.941: gdk_pixbuf_new_subpixbuf: assertion 'src_y >= 0 && src_y + height <= src_pixbuf->height' failed
Traceback (most recent call last):
  File "/app/share/com.github.hezral.clips/clips/clips_view.py", line 1088, in draw
    Gdk.cairo_set_source_pixbuf(cairo_context, final_pixbuf, x, y)
TypeError: Argument 1 does not allow None as a value

(com.github.hezral.clips:2): GdkPixbuf-CRITICAL **: 22:16:24.194: gdk_pixbuf_new_subpixbuf: assertion 'src_y >= 0 && src_y + height <= src_pixbuf->height' failed
Traceback (most recent call last):
  File "/app/share/com.github.hezral.clips/clips/clips_view.py", line 1088, in draw
    Gdk.cairo_set_source_pixbuf(cairo_context, final_pixbuf, x, y)
TypeError: Argument 1 does not allow None as a value
python3: ../../src/cairo.c:524: cairo_destroy: Assertion `CAIRO_REFERENCE_COUNT_HAS_REFERENCE (&cr->ref_count)' failed.
➜  ~ 
`

**Comment by hezral** [comment]
Fixed in [2ac1b61](https://github.com/hezral/clips/commit/2ac1b61e40d65f71e48935b05143fa6ca8f5ab7e)

Please wait for the next release or pull latest source and rebuild
